components/sup/src/http_gateway.rs

```diff
@@ -239,55 +264,42 @@ impl Server {
             };
 
             *lock.lock().expect("Control mutex is poisoned") = match bind {
-                Ok(b) => {
-                    b.start();
-                    ServerStartup::Started
-                }
-                Err(e) => {
+                Ok(_) => ServerStartup::Started,
+                Err(ref e) => {
                     error!("HTTP gateway failed to bind: {}", e);
                     ServerStartup::BindFailed
                 }
             };
 
             cvar.notify_one();
 
-            sys.run();
+            if let Ok(b) = bind {
+                b.run().expect("to start http server");
+            }
```

Comment:
Why move this block outside the lock `match` on line 266?  Why not just change `b.start();` to `b.run().expect("to start http server");`?

Reply:
Ah, unless `b.run()` is the point of no return, like `sys.run()` was before?

Reply:
Correct. That being said I am not a hug fan of matching twice on bind so if you have a better way to formulate it that would be awesome.

Reply:
I don't see a good way around it either.